node/src/index.ts

```diff
@@ -390,7 +384,7 @@ export class LocalTable<T = number[]> implements Table<T> {
    * @param indexParams The parameters of this Index, @see VectorIndexParams.
    */
   async createIndex (indexParams: VectorIndexParams): Promise<any> {
-    return tableCreateVectorIndex.call(this._tbl, indexParams)
+    return tableCreateVectorIndex.call(this._tbl, indexParams).then((newTable: any) => { this._tbl = newTable })
```

Comment:
Okay so IIUC, how this works is:

```ts
// Initializes table
const table = lance.table(...) 
// Clones the table to perform operation in another thread
const create_index_promise = table.create_index(...) 
// Another clone to perform this operation
const delete = table.delete(...) 
// Either create_index or delete will win and overwrite the table reference
await Promise.all([create_index, delete])
```

This is fine for now, since we'll disallow concurrent writes for the time being. But eventually we probably want a better model for updating the table reference after a mutation, since this is a race condition. I'll see if I can create a method soon to update a dataset to the latest version, which is probably the better way to solve this.

Reply:
https://github.com/lancedb/lance/issues/1124